File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonModel/src/main/java/triathlon/model/validator/ResultValidator.java
package triathlon.model.validator;

import triathlon.model.Result;

public class ResultValidator implements Validator<Result> {
    @Override
    public void validate(Result entity) throws ValidatorException {
        String errorMess = "";
        if (entity.getTrial() == null) {
            errorMess += "Invalid trial - cannot be null";
        }
        if (entity.getParticipant() == null) {
            errorMess += "Invalid participant - cannot be null";
        }
        if (entity.getPoints() < 0) {
            errorMess += "Invalid points - cannot be less than 0";
        }
        if (!errorMess.isEmpty()) {
            throw new ValidatorException(errorMess);
        }
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/model/validator/ResultValidator.cs
namespace model.validator;

public class ResultValidator : IValidator<Result>
{
    public void Validate(Result entity)
    {
        string errorMessage = "";
        if (entity.Points < 0)
        {
            errorMessage += "Invalid points - cannot be less than 0\n";
        }
        if (!string.IsNullOrEmpty(errorMessage))
        {
            throw new ValidatorException(errorMessage);
        }
    }
}